SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java

```diff
@@ -208,13 +197,99 @@ public DBSPAggregate createAggregate(RelNode node,
         return result;
     }
 
+    /**
+     * Given a struct type find the index of the specified field.
+     * Throws if no such field exists.
+     */
+    static int getFieldIndex(String field, RelDataType type) {
+        int index = 0;
+        for (RelDataTypeField rowField: type.getFieldList()) {
+            if (rowField.getName().equals(field))
+                return index;
+            index++;
+        }
+        throw new RuntimeException("Type " + type + " has no field named " + field);
+    }
+
     public void visitCorrelate(LogicalCorrelate correlate) {
-        throw new Unimplemented(correlate);
+        // We decorrelate queries using Calcite's optimizer.
+        // So we assume that the only correlated queries we receive
+        // are unnest-type queries.  We assume that unnest queries
```

Comment:
So we cannot have correlated subqueries?

Reply:
Yes, they are decorrelated by Calcite.